src/jit/x64/emit.dasc

```diff
@@ -1924,6 +2086,20 @@ void MVM_jit_emit_primitive(MVMThreadContext *tc, MVMJitCompiler *compiler, MVMJ
         | mov qword WORK[dst], RV;
         break;
     }
+    case MVM_OP_getcodename: {
+        MVMint16 obj  = ins->operands[0].reg.orig;
+        MVMint16 code = ins->operands[1].reg.orig;
+        | mov TMP1, aword WORK[code];
+        | cmp_repr_id TMP1, TMP2, MVM_REPR_ID_MVMCode;
+        | je >1;
+        | test_type_object TMP1;
+        | jz >1;
+        | throw_adhoc "getcodename requires a concrete object";
+        |1:
```

Comment:
This inverts the condition on the REPR ID, meaning that you'll skip the concreteness test for MVM_REPR_ID_MVMCode, and you'll execute it even if you know that it is an MVMCode object, which means that type-object MVMCode and concrete MVMArray instances (for example) will pass.